pages/components/CopyButton/index.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+
+import CopyIcon from 'pages/components/CopyIcon';
+
+import styles from './styles.module.scss';
+
+type CopyButtonProps = {
+  textToCopy: string;
+};
+
+enum CopyStates {
+  ready,
+  copying,
+  copied,
+  errored
+}
+
+const ButtonContent = {
+  [CopyStates.ready]: (<CopyIcon className={styles.copyIcon} />),
+  [CopyStates.copying]: (<div className={styles.copyCharacter}>⋯</div>),
```

Comment:
Class might make more sense as `buttonCharacter` or `buttonContent` than `copyCharacter` which is unclear.